src/collection.ts

```diff
@@ -693,8 +694,23 @@ export class Collection<TSchema extends Document = Document> {
    *
    * @param options - Optional settings for the command
    */
-  async indexInformation(options?: IndexInformationOptions): Promise<Document> {
-    return await this.indexes({ ...options, full: options?.full ?? false });
+  indexInformation(
+    options: IndexInformationOptions & { full: true }
+  ): Promise<IndexDescriptionInfo[]>;
+  indexInformation(
+    options: IndexInformationOptions & { full?: false }
+  ): Promise<IndexDescriptionCompact>;
+  indexInformation(
+    options: IndexInformationOptions & { full: boolean }
+  ): Promise<IndexDescriptionCompact | IndexDescriptionInfo[]>;
+  indexInformation(): Promise<IndexDescriptionCompact>;
```

Comment:
I was testing a bit more and I think we're missing one last overload.  Currently, the following has a type error:

```typescript
async function info(options: IndexInformationOptions) {
  const ret = await collection.indexInformation(options);
}

No overload matches this call.
  Overload 1 of 4, '(options: IndexInformationOptions & { full: true; }): Promise<IndexDescriptionInfo[]>', gave the following error.
    Argument of type 'IndexInformationOptions' is not assignable to parameter of type 'IndexInformationOptions & { full: true; }'.
      Type 'IndexInformationOptions' is not assignable to type '{ full: true; }'.
        Types of property 'full' are incompatible.
          Type 'boolean' is not assignable to type 'true'.
  Overload 2 of 4, '(options: IndexInformationOptions & { full?: false; }): Promise<IndexDescriptionCompact>', gave the following error.
    Argument of type 'IndexInformationOptions' is not assignable to parameter of type 'IndexInformationOptions & { full?: false; }'.
      Type 'IndexInformationOptions' is not assignable to type '{ full?: false; }'.
        Types of property 'full' are incompatible.
          Type 'boolean' is not assignable to type 'false'.
  Overload 3 of 4, '(options: IndexInformationOptions & { full: boolean; }): Promise<IndexDescriptionInfo[] | IndexDescriptionCompact>', gave the following error.
    Argument of type 'IndexInformationOptions' is not assignable to parameter of type 'IndexInformationOptions & { full: boolean; }'.
      Type 'IndexInformationOptions' is not assignable to type '{ full: boolean; }'.
        Property 'full' is optional in type 'IndexInformationOptions' but required in type '{ full: boolean; }'.ts(2769)
```

It seems like the problem is that the overload that matches a generic `IndexInformationOptions` contains a boolean literal (`{ full?: false }`), but because we don't have a const boolean, no overload is matched.

If we removed the `IndexInformationOptions & { full?: boolean }` overload and instead provided an overload:
```
indexInformation(
    options: IndexInformationOptions
  ): Promise<IndexDescriptionCompact | IndexDescriptionInfo[]>;
```

The issue is resolved.   This has the downside of losing the overload which provides a default return type when and empty options object is provided.

Reply:
I figured why this happens. This type `IndexInformationOptions & { full: boolean }` has the `full` property required and that makes `IndexInformationOptions` unassignable. Nice catch, I'll add a test case for this.